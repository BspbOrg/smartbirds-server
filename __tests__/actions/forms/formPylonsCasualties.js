/* eslint-env node, jest */
/* globals setup */

const { omit } = require('lodash')

describe('Action: fromPylonsCasualties.create', () => {
  const action = 'formPylonsCasualties:create'

  const bareRecord = {
    species: 'acepiter',
    count: 1,
    age: {
      label: {
        en: 'Ad.',
        bg: 'Ad.'
      }
    },
    sex: {
      label: {
        en: 'Male',
        bg: 'Мъжки'
      }
    },
    causeOfDeath: {
      label: {
        en: 'Electrocution',
        bg: 'Токов удар'
      }
    },
    bodyCondition: {
      label: {
        en: 'Fresh corpse',
        bg: 'Свеж труп'
      }
    },
    habitat100mPrime: {
      label: {
        en: 'Arable land',
        bg: 'Обработваеми площи'
      }
    },
    habitat100mSecond: {
      label: {
        en: 'Grassland',
        bg: 'Открити затревени площи'
      }
    },
    startDateTime: '2022-02-08T08:10Z',
    endDateTime: '2022-02-08T10:15Z',
    latitude: 42.1463749,
    longitude: 24.7492006,
    monitoringCode: 'pylon_mon_code',
    observationDateTime: '2022-02-08T10:15Z'
  }

  afterEach(async () => {
    await setup.api.models.formPylonsCasualties.destroy({
      force: true,
      where: {}
    })
  })

  setup.describeAsAuth((runAction) => {
    test('can create with default values', async () => {
      const response = await runAction(action, bareRecord)
      expect(response.error).toBeFalsy()
      expect(response.data).toEqual(expect.objectContaining({
        id: expect.any(Number)
      }))
    })

    test.each(['species', 'count', 'causeOfDeath'])('cannot create without %s', async (field) => {
      const response = await runAction(action, omit(bareRecord, field))
      expect(response.error).toMatchSnapshot('Missing param')
    })

    test.each(['age', 'sex', 'bodyCondition', 'habitat100mPrime', 'habitat100mSecond'])('can create without %s', async (field) => {
      const response = await runAction(action, omit(bareRecord, field))
      expect(response.error).toBeFalsy()
      expect(response.data).toEqual(expect.objectContaining({
        id: expect.any(Number)
      }))
    })
  })
})
