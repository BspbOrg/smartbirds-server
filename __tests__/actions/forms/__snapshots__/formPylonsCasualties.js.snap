// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action: fromPylonsCasualties.create admin cannot create without causeOfDeath: Missing param 1`] = `"Error: actionhero.errors.missingParams"`;

exports[`Action: fromPylonsCasualties.create admin cannot create without count: Missing param 1`] = `"Error: actionhero.errors.missingParams"`;

exports[`Action: fromPylonsCasualties.create admin cannot create without species: Missing param 1`] = `"Error: actionhero.errors.missingParams"`;

exports[`Action: fromPylonsCasualties.create birds moderator cannot create without causeOfDeath: Missing param 1`] = `"Error: actionhero.errors.missingParams"`;

exports[`Action: fromPylonsCasualties.create birds moderator cannot create without count: Missing param 1`] = `"Error: actionhero.errors.missingParams"`;

exports[`Action: fromPylonsCasualties.create birds moderator cannot create without species: Missing param 1`] = `"Error: actionhero.errors.missingParams"`;

exports[`Action: fromPylonsCasualties.create cbm moderator cannot create without causeOfDeath: Missing param 1`] = `"Error: actionhero.errors.missingParams"`;

exports[`Action: fromPylonsCasualties.create cbm moderator cannot create without count: Missing param 1`] = `"Error: actionhero.errors.missingParams"`;

exports[`Action: fromPylonsCasualties.create cbm moderator cannot create without species: Missing param 1`] = `"Error: actionhero.errors.missingParams"`;

exports[`Action: fromPylonsCasualties.create user cannot create without causeOfDeath: Missing param 1`] = `"Error: actionhero.errors.missingParams"`;

exports[`Action: fromPylonsCasualties.create user cannot create without count: Missing param 1`] = `"Error: actionhero.errors.missingParams"`;

exports[`Action: fromPylonsCasualties.create user cannot create without species: Missing param 1`] = `"Error: actionhero.errors.missingParams"`;
